erpnext/hr/doctype/payroll_entry/payroll_entry.py

```diff
@@ -341,36 +417,93 @@ def make_accrual_jv_entry(self):
 
 		return jv_name
 
-	def make_payment_entry(self):
+	def payroll_entry_has_bank_entries(self):
+
+		journal_entries_amount_sum = frappe.db.sql(
+			'select sum(debit-credit) as amount_sum from `tabJournal Entry Account` '
+			'where reference_type="Payroll Entry" '
+			'and reference_name=%s and docstatus=1', self.name, as_list=True
+		)
+
+		journal_entries_amount_sum = flt(journal_entries_amount_sum[0][0]) if journal_entries_amount_sum else 0
+		salary_slips = self.get_salary_slips_for_payment()
+
+		ss_rounded_total_sums = sum([ss.rounded_total for ss in salary_slips])
+
+		return 0 if journal_entries_amount_sum < ss_rounded_total_sums else 1
+
+	def get_salary_slips_for_payment(self, include_draft=False):
+		docstatus_cond = "t1.docstatus < 2" if include_draft else "t1.docstatus = 1"
+		filter_cond = self.get_filter_condition()
+
+		salary_slips = frappe.db.sql("""
+			select t1.name, t1.employee, t1.salary_mode, t1.bank_name, t1.bank_account_no, t1.net_pay, t1.rounded_total
+			from `tabSalary Slip` t1
+			where {0} and start_date >= %s and end_date <= %s {1}
+		""".format(docstatus_cond, filter_cond), (self.start_date, self.end_date), as_dict=True)
+
+		return salary_slips
+
+	def get_disbursement_mode_details(self):
+		salary_slips = self.get_salary_slips_for_payment()
+		salary_modes = set([ss.salary_mode for ss in salary_slips if ss.salary_mode and ss.salary_mode != "Cheque"])
+		bank_names = set([ss.bank_name for ss in salary_slips if ss.salary_mode == 'Bank' and ss.bank_name])
+
+		return list(salary_modes), list(bank_names)
+
+	def get_bank_details(self):
+		sal_slips_in_payroll = self.get_salary_slips_for_payment(include_draft=True)
+		sal_slips_in_payroll = [d for d in sal_slips_in_payroll if d.bank_name]
```

Comment:
Change condition to `d.salary_mode == 'Bank' and d.bank_name`
Make sure that salary_mode is provided with the `get_salary_slips_for_payment` method